packages/mendel-middleware/index.js

```diff
@@ -106,7 +106,17 @@ function MendelMiddleware(opts) {
         var pack = bpack({raw: true, hasExports: true});
         var decodedResults = trees.findTreeForHash(params.bundle, params.hash);
         if (!decodedResults || decodedResults.error) {
-            return notFound(res, decodedResults && decodedResults.error);
+            return notFound(res, decodedResults && decodedResults.error, params);
+        }
+
+        var modules = indexedDeps(decodedResults.deps.filter(Boolean));
+
+        if (!modules.length) {
+            // Something wrong, modules shouldn't be zero
+            return notFound(res, {
+                code: 'EMPTYBUNDLE',
+                message: 'Tree contents are empty'
+            }, params);
```

Comment:
Getting rid of `prune` and getting rid on `shasum` and just having `normalizedId` is exactly the same. But when you remove prune you can accidentally include extra entire files. One more reason to fix `mendel-generator-prune` to remove the renaming index part.